src/core/EntrySearcher.cpp

```diff
@@ -161,26 +151,8 @@ bool EntrySearcher::searchEntryImpl(const Entry* entry)
     // Build a group hierarchy to allow searching for e.g. /group1/subgroup*
     auto hierarchy = entry->group()->hierarchy().join('/').prepend("/");
 
-    QList<SearchTerm> terms;
-    if (m_skipProtected) {
-        for (const auto& term : m_searchTerms) {
-            bool skip = term.field == Field::Password
-                        || (term.field == Field::AttributeValue && entry->attributes()->isProtected(term.word));
-            if (skip) {
-                continue;
-            }
-            terms << term;
-        }
-        // empty search terms match nothing
-        if (terms.isEmpty()) {
-            return false;
-        }
-    } else {
-        terms = m_searchTerms;
-    }
-
     bool found;
```

Comment:
`found` may be uninitialized if every term in `m_searchTerms` are protected and `m_skipProtected == true`. However, neither initializing it to `true` or `false` works in all cases.

Consider a database with two entries:
* entry 1: title "Apple", password "pass"
* entry 2: title "Banana", password "word"

If `found` initializes to `true`, a search query "password:pass" will return entry 1, while it should return nothing

If `found` initializes to `false`, a search query "password:pass title:Apple" returns nothing, but it should return entry 1.

Basically, all terms matching protected attributes should be removed first to not affecting the shortcut logic.

Reply:
Good point, then we should continue instead of break to skip the short circuit logic and default found to false.

Reply:
I split the logic of `term.exclude` handling and short circuit. It's now cleaner IMHO especially with the addition of `skipProtected` to the mix.

I also added my above comment as a new test case.